### PR TITLE
ML-055/improve elasticsearch

### PR DESCRIPTION
**Pull request by WhatWasThatt**
# Description

add elasticsearch data importer
add search controller
add search service
update searchable module
add indexes, types and analyzers to movie, person and actor_role models
add rake task for elasticsearch importer

## Trello card link

https://trello.com/c/AGkyXF8o

# Checklist:

- [x] My code fixes the problem or adds required functionality
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove that my feature works
- [x] New and existing unit tests pass locally with my changes

# Additional requirements:

- [ ] Does your code require any migrations to be ran?
- [ ] Does your code require installing any new gems or system dependencies?
- [ ] Does your code require any changes to current deployment process?

# Additional Information

### CODE REVIEW

File: app/services/search/results/base_builder.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module Search
+  module Results
+    class BaseBuilder
+      def initialize(record)
+        @record = record
+      end
+
+      def autocomplete_hint
+        'Not implement yet'
```
Comment (twill11) [inline app/services/search/results/base_builder.rb:9]: i think u should add a raise 'NotImplemented', what do you think?

File: app/services/search/results/hint_builder.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        new(record).call
+      end
+
+      def call
+        result_builder.autocomplete_hint
+      end
+
+      private
+
+      attr_accessor :record
```
Comment (twill11) [inline app/services/search/results/hint_builder.rb:18]: do u actually need accessor?

File: app/services/search/autocompleter.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+          hint: build_hint(result),
+          record_type: result.class.name,
+          record_id: result.id
+        }
+      end
+    end
+
+    private
+
+    attr_accessor :query
```
Comment (twill11) [inline app/services/search/autocompleter.rb:25]: same

File: app/services/search/results/base_builder.rb (hunk 1 above)
Comment (WhatWasThatt) [reply app/services/search/results/base_builder.rb]: sounds good

File: app/services/search/results/hint_builder.rb (hunk 2 above)
Comment (WhatWasThatt) [reply app/services/search/results/hint_builder.rb]: you're right

**Review by twill11 — APPROVED** [review]
gj!

**Review by dmitrymalllpr — DISMISSED** [review]

**Review by dmitrymalllpr — APPROVED** [review]

**Review by KarinaKlevzhits — APPROVED** [review]

**Pull request merged**